### PR TITLE
Replace asserts

### LINKED ISSUE #263: Create exceptions and replace all asserts
## Description
Currently, we use asserts for things like input/response validation. Those asserts would be compiled out if you want to generate optimized bytecode.

Replace all asserts with custom made ```Exceptions``` (those exceptions should be placed in a single file in our repo).

## Expected Behavior
Have a file with all the custom made exceptions and all asserts replaced with them

### PR DESCRIPTION
**Pull request by clovis1122**
## Description
Replaces `asserts` statements and instead raises exceptions. I only found one assertion that is not inside the `testsuite` folder.

Follow up: should I create a custom `JwtException` class? I didn't do it since its just one...

Fixes #263

## Testing
-

## Checklist
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/vmck/acs-interface/labels)
- [x] Tests (if they apply)

### CODE REVIEW

**Comment by gmuraru** [comment]
Aren't there more "asserts" or is only this one?

**Comment by clovis1122** [comment]
Only one I found @gmuraru, the rest are on the testsuite folder.

image

**Comment by gmuraru** [comment]
I think it is ok like this. I was expecting that it might be more and making custom exceptions for those might make sense - for now it is ok :+1: 
Could you also add a quick test where you call done and you expect and exception?

**Comment by clovis1122** [comment]
Hey @gmuraru, so while creating the test, I realized returning an HTTP 400 response is more appropriate in this case, to let the one calling the API know that there's a mistake with the token. I added a happy path test, as well as testing this HTTP 400 response. Let me know if this is ok.

**Comment by gmuraru** [comment]
Yep, from my side, it looks great! Waiting for @Jokeswar to give the second approval.

File: testsuite/test_views.py
Comment (gmuraru) [inline testsuite/test_views.py:109]: Love this! From the Flintstones Family, right?

**Review by gmuraru — APPROVED** [review]

**Comment by Jokeswar** [comment]
@clovis1122 if you could run [black formatter](https://github.com/psf/black) and [flake8](https://pypi.org/project/flake8/) before we merge and fix any errors that flake8 might report, it would be great.

**Comment by gmuraru** [comment]
@clovis1122 is black formatter or flake8 reporting any problems?

**Comment by clovis1122** [comment]
I'll try running it later today. Sorry for the delay!

**Comment by gmuraru** [comment]
No problem, thank you for the contribution :D

**Comment by clovis1122** [comment]
Done, please check the last commit  @Jokeswar @gmuraru

**Comment by gmuraru** [comment]
LGTM! @Jokeswar what do you think?

**Review by Jokeswar — APPROVED** [review]

**Pull request merged**
